File: solutions/java/src/onlinestockbrokeragesystem/Stock.java
package onlinestockbrokeragesystem;

public class Stock {
    private final String symbol;
    private final String name;
    private double price;

    public Stock(String symbol, String name, double price) {
        this.symbol = symbol;
        this.name = name;
        this.price = price;
    }

    public synchronized void updatePrice(double newPrice) {
        price = newPrice;
    }

    public String getSymbol() {
        return symbol;
    }

    public String getName() {
        return name;
    }

    public double getPrice() {
        return price;
    }
}

File: solutions/csharp/onlinestockbrokeragesystem/Stock.cs
namespace OnlineStockBrokerageSystem
{
    public class Stock
    {
        public string Symbol { get; }
        public string Name { get; }
        public double Price { get; private set; }

        public Stock(string symbol, string name, double price)
        {
            Symbol = symbol;
            Name = name;
            Price = price;
        }

        public void UpdatePrice(double newPrice)
        {
            Price = newPrice;
        }
    }
}